gateway-server/src/main/java/org/apache/knox/gateway/services/token/impl/JDBCTokenStateService.java

```diff
@@ -224,6 +224,20 @@ protected void evictExpiredTokens() {
     } catch (SQLException e) {
       log.errorRemovingTokensFromDatabase(e.getMessage(), e);
     }
+
+    //removing disabled KnoxSSO cookies since they are no longer needed
+    try {
+      final Set<String> disabledKnoxSsoCookies = tokenDatabase.getDisabledKnoxSsoCookies();
+      if (!disabledKnoxSsoCookies.isEmpty()) {
+        log.removingDisabledKnoxSsoCookiesFromDatabase(disabledKnoxSsoCookies.size(),
+            String.join(", ", disabledKnoxSsoCookies.stream().map(tokenId -> Tokens.getTokenIDDisplayText(tokenId)).collect(Collectors.toSet())));
+        for (String tokenId : disabledKnoxSsoCookies) {
```

Comment:
How many tokens do we expect to have normally? If it's a lot then maybe having a dedicated sql delete statement would be better.

Reply:
Good argument. I also had the same idea but I wanted to have the PR out ASAP so that you guys can review it. Let me submit a new PS with the updated `DELETE` statement.